libvcx/src/libindy/proofs/proof_request_internal.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct Filter {
+    pub schema_id: Option<String>,
+    pub schema_issuer_did: Option<String>,
+    pub schema_name: Option<String>,
+    pub schema_version: Option<String>,
+    pub issuer_did: Option<String>,
+    pub cred_def_id: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(untagged)]
+pub enum Restrictions {
+    V1(Vec<Filter>),
+    V2(::serde_json::Value),
+}
```

Comment:
Do we need to use both?

Reply:
Honestly, I don't know. It doesn't seem that any of them is directly used in code, but I suppose you can get proof request in 2 different formats which needs to be deserialized.

Reply:
What determines the proof request format? Let's try to use only V1 / V2 and see what breaks :)